lib/DocblockParser/Tests/Unit/ParserTest.php

```diff
@@ -45,4 +45,39 @@ public function provideParse(): Generator
             ])
         ];
     }
+
+    /**
+     * @dataProvider provideMultidimensionalType
+     */
+    public function testParseMultidimensionalArray(string $type, ?string $expectedType = null): void
```

Comment:
the expected way to test this is via. the files in: `Tests/Unit/Printer/examples/*.test`

just add:

```
/**
 * @var array
 */
```

and run the test, it shold automatically complete the test by printing the AST (it's basically a snapshot test):

```
/**
 * @var array
 */
---
Docblock: = 
 ElementList: = /**
 * 
  VarTag: = @var
   ArrayNode: = array
 */
```